src/components/main/friends/sidebar/mod.rs

```diff
@@ -145,45 +152,57 @@ pub fn FindFriends(cx: Scope, account: Account, add_error: UseState<String>) ->
     let codeCopied = { l.code_copied.to_string() };
     let account2 = account.clone();
 
+    let search_results = use_state(&cx, Vec::<SelectOption>::new);
+
+    let on_change = move |ev: FormEvent| {
+        let value = ev.data.value.clone();
+
+        remote_friend.set(value.clone());
+
+        if value.len() < 3 {
+            if !search_results.is_empty() {
+                search_results.set(Vec::new());
+            }
+
+            return;
+        }
+
+        match account.get_identity(Identifier::user_name(value.clone().as_str())) {
```

Comment:
No need for the additional clone. You can also do `let results = account.get_identity(Identifier::user_name(value.as_str())).unwrap_or_default();` and possibly extend over that with the iterator to add `SelectOption` since a successful or failed lookup would yield a error (although there is still a chance for it to return `Err` too but unlikely if multipass is initialized properly)

Reply:
updated according to your suggestions, thanks